### PR TITLE
add some more deprecation warnings to cluster

### PR DESCRIPTION
**Pull request by mikedanese**
Part of https://github.com/kubernetes/kubernetes/issues/49213
@kubernetes/sig-cluster-lifecycle-misc

### CODE REVIEW

File: cluster/kube-up.sh [hunk 1]
```diff
@@ -32,6 +32,31 @@ fi
...
+  "windows"
+)
+
+for provider in "${DEPRECATED_PROVIDERS[@]}"; do
+  if [[ "${KUBERNETES_PROVIDER}" == "${provider}" ]]; then
+    cat <<EOF 1>&2
+
+!!! DEPRECATION NOTICE !!!
+
+The '${provider}' kube-up provider is deprecated and will be removed in a future
```
Comment (pipejakob) [inline cluster/kube-up.sh]: Is there an issue or announcement we can link to here to provide more context about the deprecation? Also, if someone sees this message and immediately panics and wants to convince us **not** to remove this support, how should they proceed? A little clarity up front will help us organize any support pings we get in the future.

Comment (luxas) [inline cluster/kube-up.sh]: What @pipejakob 
Also we could state the timeline up front here (removal in v1.9?)

Comment (luxas) [inline cluster/kube-up.sh]: @mikedanese sounds reasonable to defer that.

**Pull request merged**